File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {return false;}

        char[] s_letters = s.toCharArray();
        char[] t_letters = t.toCharArray();

        Arrays.sort(s_letters);
        Arrays.sort(t_letters);

        return Arrays.equals(s_letters,t_letters);
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_counts, t_counts = Counter(s), Counter(t)
        
        for count in s_counts:
            if count in t_counts:
                if s_counts[count] != t_counts[count]:
                    return False
            else:
                return False
        
        if len(s) != len(t):
            return False
        
        return True